setup/build_compile.sh

```diff
@@ -0,0 +1,187 @@
+#!/bin/bash
+
+# Build GMP
+function build_gmp() {
```

Comment:
I did the best I could to not repeat code, but now the script is even longer (and in my opinion, uglier) than it was before...